JoyboyCommunity/src/components/modal/index.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import {Portal} from 'react-native-portalize';
+
+import Line from '../../../assets/svgs/svgComponents/Line';
+import {
+  ButtonText,
+  Container,
+  Content,
+  IconContainer,
+  ModalContainer,
+  Overlay,
+  StyledButton,
+  Title,
+} from './styled';
+
+interface Button {
+  label: string;
+  type: 'dangerous' | 'primary' | 'secondary';
+  onPress: () => void;
```

Comment:
No need to create an interface for button

Reply:
The component is a Touchableopacity and not a button

Reply:
I'm addressing the `interface Button` it should be removed and moved into the props directly.

```ts
export type CustomModalProps = {
  name: string;
  buttons: {
    label: string;
    type: 'dangerous' | 'primary' | 'secondary';
    onPress: () => void;
  }[];
  icon: React.ReactNode;
  description: string;
  visible: boolean;
}
```